meson.build

```diff
@@ -266,43 +266,30 @@ cc = meson.get_compiler('c')
 flags = ['-D_FILE_OFFSET_BITS=64']
 link_flags = []
 
-test_flags = ['-Werror=implicit-function-declaration',
-              '-Wno-missing-field-initializers',
-              '-Wno-sign-compare',
-              '-Wno-unused-parameter',
-              '-Wno-cast-function-type',
+test_flags = ['-Wdisabled-optimization',
               '-Wempty-body',
-              '-Wdisabled-optimization',
+              '-Wformat',
+              '-Wimplicit-fallthrough',
+              '-Wmissing-prototypes',
+              '-Wparentheses',
+              '-Wpointer-arith',
+              '-Wshadow',
               '-Wstrict-prototypes',
-              '-Wno-format-zero-length',
-              '-Wno-redundant-decls',
+              '-Wundef',
               '-Wvla',
-              '-Wno-format-truncation',
-              '-Wimplicit-fallthrough',
+              '-Werror=format-security',
+              '-Werror=implicit-function-declaration',
+              '-Wno-cast-function-type',
+              '-Wno-format-zero-length',
+              '-Wno-missing-field-initializers',
+              '-Wno-pointer-sign',
+              '-Wno-sign-compare',
+              '-Wno-switch',
+              '-Wno-unused-parameter',
               '-fno-math-errno']
 
 flags += cc.get_supported_arguments(test_flags)
 
-if cc.has_multi_arguments('-Wformat', '-Werror=format-security')
-    flags += ['-Wformat', '-Werror=format-security']
-endif
```

Comment:
`'-Werror=format-security'` requires the `has_multi_arguments` test. It doesn't work with `get_supported_arguments` which only supports one.

Reply:
Just GCC things. I see that it errors out when it is used alone. Will update.